.github/workflows/uberjar.yml

```diff
@@ -43,19 +43,23 @@ jobs:
       oss_exists: ${{ steps.check-oss.outcome == 'success' }}
       should_build: ${{ steps.decide.outputs.should_build }}
     steps:
-      - name: Try to download existing EE artifact
+      - name: Check out the code
+        uses: actions/checkout@v4
+      - name: Try to fetch existing EE artifact
         id: check-ee
         continue-on-error: true
-        uses: actions/download-artifact@v4
+        uses: ./.github/actions/fetch-artifact
```

Comment:
There is nothing in `fetch-artifact` that tells it which run to pull the artifact from, it just pulls the latest. So this is going to succeed even if it shouldn't. This is probably more a bug in `fetch-artifact` where it should be using the run id.